### ISSUE TITLE
user module fails with DeprecationWarning when using -c local

### ISSUE THREAD

**Issue by roller** [ansible/ansible#1626]
I'm having trouble running the user module on the latest git devel branch with the local connection.  It appears to work fine, but get immediately followed by a deprecation warning.

I have no idea why this doesn't show up over an ssh connection.

This is with the latest debian stable.

$ python --version
Python 2.6.6

```
# ansible -vvv localhost -i inventory/hosts.ini -c local -m user -a name=testuser 
<localhost> EXEC mkdir -p /var/tmp/ansible-1352931409.76-49800360366093 && echo /var/tmp/ansible-1352931409.76-49800360366093
<localhost> REMOTE_MODULE user name=testuser
<localhost> PUT /tmp/tmpuZKf7W TO /var/tmp/ansible-1352931409.76-49800360366093/user
<localhost> EXEC /usr/bin/python /var/tmp/ansible-1352931409.76-49800360366093/user; rm -rf /var/tmp/ansible-1352931409.76-49800360366093/ >/dev/null 2>&1
localhost | FAILED => failed to parse: {"comment": "", "shell": "/bin/sh", "group": 1006, "name": "testuser", "changed": false, "state": "present", "home": "/home/testuser", "append": false, "uid": 998}
/var/tmp/ansible-1352931409.76-49800360366093/user:1031: DeprecationWarning: object.__new__() takes no parameters
  return super(cls, subclass).__new__(subclass, *args, **kwargs)
```

A work around patch is as follows, but I don't know if that's the proper approach to fix the warning.  

https://gist.github.com/4075370

**Comment by mpdehaan** [comment]
Hmm, yes.  I believe we need to leave off the args/kwargs stuff in this
case.

**Comment by mpdehaan** [comment]
fix pushed in fdafb78

the short answer about the remote is that there is various logic to filter out random non-JSON noise (like 'tcgetattr' messages that is probably eating the message.
